--- a/nss_ifb.c
+++ b/nss_ifb.c
@@ -259,11 +259,11 @@ static int __init nss_ifb_init_module(vo
 	struct net_device *dev;
 	int err;
 
-	down_write(&pernet_ops_rwsem);
-	rtnl_lock();
-	err = __rtnl_link_register(&nss_ifb_link_ops);
+	err = rtnl_link_register(&nss_ifb_link_ops);
 	if (err < 0)
-		goto out;
+		return err;
+
+	rtnl_net_lock(&init_net);
 
 	dev = alloc_netdev(sizeof(struct nss_ifb_dev_private), "nssifb",
 			NET_NAME_UNKNOWN, nss_ifb_setup);
@@ -271,17 +271,14 @@ static int __init nss_ifb_init_module(vo
 	if (dev) {
 		dev->rtnl_link_ops = &nss_ifb_link_ops;
 		err = register_netdevice(dev);
-	}
-	else {
+	} else {
 		err = -ENOMEM;
 	}
 
-	if (err)
-		__rtnl_link_unregister(&nss_ifb_link_ops);
+	rtnl_net_unlock(&init_net);
 
-out:
-	rtnl_unlock();
-	up_write(&pernet_ops_rwsem);
+	if (err)
+		rtnl_link_unregister(&nss_ifb_link_ops);
 
 	if (!err)
 		pr_info("NSS IFB module loaded.\n");
@@ -301,4 +298,5 @@ static void __exit nss_ifb_cleanup_modul
 module_init(nss_ifb_init_module);
 module_exit(nss_ifb_cleanup_module);
 MODULE_LICENSE("GPL");
+MODULE_INFO(description, "NSS IFB Interface");
 MODULE_ALIAS_RTNL_LINK("nss_ifb");
