--- a/mc_netlink.c
+++ b/mc_netlink.c
@@ -776,7 +776,7 @@ static void __mc_netlink_receive(struct
 					mc->local_query_interval = i->val * HZ;
 					if (timer_pending(&mc->qtimer) ?
 						time_after(mc->qtimer.expires, jiffies + mc->local_query_interval) :
-							try_to_del_timer_sync(&mc->qtimer) >= 0) {
+							timer_delete_sync_try(&mc->qtimer) >= 0) {
 						if (mc->started)
 							mod_timer(&mc->qtimer, jiffies + mc->local_query_interval);
 					}
--- a/mc_wifi_events.c
+++ b/mc_wifi_events.c
@@ -24,6 +24,7 @@
 #include "mc_netlink.h"
 #include "mc_snooping.h"
 #include "mc_forward.h"
+#include "mc_wifi_events.h"
 #include <linux/in6.h>
 
 struct sta_leave_event {
@@ -81,7 +82,7 @@ static int mc_event_recv(struct socket *
  * mc_sta_leave_event()
  *	Delete the mac entry of sta from bridge snooping table
  */
-void mc_sta_leave_event(struct net_device *dev,__u8 *from_mac)
+static void mc_sta_leave_event(struct net_device *dev,__u8 *from_mac)
 {
 
 	struct mc_struct *mc;
@@ -154,7 +155,7 @@ void mc_sta_leave_event(struct net_devic
  * mc_process_sta_leave_event()
  * 	Parse and process sta_leave event
  */
-void mc_process_sta_leave_event(struct nlmsghdr *nlh, int cmd)
+static void mc_process_sta_leave_event(struct nlmsghdr *nlh, int cmd)
 {
 	struct nlattr **tb = NULL;
 	uint8_t mac[ETH_ALEN] = {0};
@@ -438,7 +439,7 @@ void mc_handle_vbss_event(struct net_dev
 /* mc_process_vendor_event()
  *	Process the vendor events
  */
-void mc_process_vendor_event(struct nlmsghdr *nlh, int cmd)
+static void mc_process_vendor_event(struct nlmsghdr *nlh, int cmd)
 {
 	struct net_device *dev;
 	u_int32_t ifidx;
@@ -557,7 +558,7 @@ static int mc_event_handler(void *buf, i
  * 	Parse the generic control family response message and return
  * 	the multicast id of MLME and VENDOR multicast groups of nl80211 family.
  */
-int mc_get_event_id(struct nlmsghdr *nlh, const char *event_name, int *event_id)
+static int mc_get_event_id(struct nlmsghdr *nlh, const char *event_name, int *event_id)
 {
     struct nlattr *tb[CTRL_ATTR_MAX+1];
     struct nlattr *mcgrp;
@@ -608,7 +609,7 @@ int mc_get_event_id(struct nlmsghdr *nlh
  *	Get the multicast id of MLME and VENDOR multicast groups of nl80211 family.
  */
 
-int mc_parse_genl_ctrl_response(struct nlmsghdr *nlh, int *mlme_id, int *vendor_id)
+static int mc_parse_genl_ctrl_response(struct nlmsghdr *nlh, int *mlme_id, int *vendor_id)
 {
 	int res_mlme, res_vendor;
 
@@ -642,7 +643,7 @@ int mc_parse_genl_ctrl_response(struct n
  *	Construct a message to generic control family to resolve the
  *	nl80211 multicast groups.
  */
-struct nlmsghdr *mc_construct_nl_message(void)
+static struct nlmsghdr *mc_construct_nl_message(void)
 {
 	int flags = GFP_ATOMIC;
 	struct sk_buff *skb;
@@ -705,7 +706,7 @@ err:
  *	Resolve the nl80211 family multicast groups to
  *	receive Wi-Fi specific events.
  */
-int mc_resolve_nl80211_family(struct socket *sock, struct sockaddr_nl *addr)
+static int mc_resolve_nl80211_family(struct socket *sock, struct sockaddr_nl *addr)
 {
 	struct nlmsghdr *nlh;
 	struct kvec iov = {0};
--- a/mc_snooping.c
+++ b/mc_snooping.c
@@ -645,7 +645,7 @@ static void mc_mdb_destroy(struct mc_mdb
 	}
 
 	hlist_del_rcu(&mdb->hlist);
-	del_timer_sync(&mdb->etimer);
+	timer_delete_sync(&mdb->etimer);
 
 	if (mdb->flood_ifcnt || mdb->encap_dev_cnt) {
 		mdb->flood_ifcnt = 0;
@@ -677,7 +677,7 @@ static void mc_agingtimer_reset(struct m
 
 	if (timer_pending(&mc->agingtimer) ?
 			time_after(mc->agingtimer.expires, jiffies + expires) :
-				try_to_del_timer_sync(&mc->agingtimer) >= 0) {
+				timer_delete_sync_try(&mc->agingtimer) >= 0) {
 		mod_timer(&mc->agingtimer, jiffies + expires);
 		MC_PRINT("Reset Group Membership Interval ageing timer, expires = %u\n",
 				jiffies_to_msecs(expires) / 1000);
@@ -705,7 +705,7 @@ static void mc_rtimer_reset(struct mc_st
 
 	if (timer_pending(&mc->rtimer) ?
 			time_after(mc->rtimer.expires, jiffies + expires) :
-				try_to_del_timer_sync(&mc->rtimer) >= 0) {
+				timer_delete_sync_try(&mc->rtimer) >= 0) {
 		mod_timer(&mc->rtimer, jiffies + expires);
 		MC_PRINT("Reset Querier Interval ageing timer, expires = %u\n",
 				jiffies_to_msecs(expires) / 1000);
@@ -715,17 +715,9 @@ static void mc_rtimer_reset(struct mc_st
 /* mc_mdb_expired
  *	mdb expired
  */
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 4, 0))
 static void mc_mdb_expired(struct timer_list *tl)
-#else
-static void mc_mdb_expired(unsigned long data)
-#endif
 {
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 4, 0))
-	struct mc_mdb_entry *mdb = from_timer(mdb, tl, etimer);
-#else
-	struct mc_mdb_entry *mdb = (struct mc_mdb_entry *)data;
-#endif
+	struct mc_mdb_entry *mdb = timer_container_of(mdb, tl, etimer);
 	struct mc_struct *mc = mdb->mc;
 	struct mc_port_group *pg;
 	struct hlist_node *pgh;
@@ -759,11 +751,7 @@ static struct mc_mdb_entry *mc_mdb_creat
 		mdb->group = *group;
 		mdb->mc = mc;
 		mdb->ageing_query = jiffies;
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 4, 0))
 		timer_setup(&mdb->etimer, mc_mdb_expired, 0);
-#else
-		setup_timer(&mdb->etimer, mc_mdb_expired, (unsigned long)mdb);
-#endif
 		hlist_add_head_rcu(&mdb->hlist, head);
 	} else {
 		MC_PRINT("%s: No memory\n", __func__);
@@ -1215,7 +1203,7 @@ static void mc_leave_group(struct mc_str
 	fg->ageing_timer = now - expire_time + expire_delay;
 	if (timer_pending(&mc->agingtimer) ?
 		time_after(mc->agingtimer.expires, jiffies + expire_delay) :
-		try_to_del_timer_sync(&mc->agingtimer) >= 0) {
+		timer_delete_sync_try(&mc->agingtimer) >= 0) {
 		mod_timer(&mc->agingtimer, jiffies + expire_delay);
 		MC_PRINT("Reset Group Membership Interval ageing timer, expires = %u ms\n",
 			leave_delay_msecond);
@@ -2082,7 +2070,7 @@ static void mc_query_cycle_start(struct
 			goto out;
 		if (timer_pending(&mdb->etimer) ?
 				time_after(mdb->etimer.expires, now + root_qe->max_resp_time) :
-				try_to_del_timer_sync(&mdb->etimer) >= 0) {
+				timer_delete_sync_try(&mdb->etimer) >= 0) {
 			mdb->timer_base = now;
 			mod_timer(&mdb->etimer, now + root_qe->max_resp_time);
 			mc_query_timer_dump(mdb, root_qe->max_resp_time, 1);
@@ -2097,7 +2085,7 @@ static void mc_query_cycle_start(struct
 					continue;
 				if (timer_pending(&mdb->etimer) ?
 						time_after(mdb->etimer.expires, now + root_qe->max_resp_time) :
-						try_to_del_timer_sync(&mdb->etimer) >= 0) {
+						timer_delete_sync_try(&mdb->etimer) >= 0) {
 					mdb->timer_base = now;
 					mod_timer(&mdb->etimer, now + root_qe->max_resp_time);
 					mc_query_timer_dump(mdb, root_qe->max_resp_time, 0);
@@ -2948,10 +2936,10 @@ static int mc_stop(struct mc_struct *mc)
 	mc_mdb_flush(mc);
 	mc_rlist_flush(mc);
 
-	del_timer_sync(&mc->qtimer);
-	del_timer_sync(&mc->agingtimer);
-	del_timer_sync(&mc->rtimer);
-	del_timer_sync(&mc->evtimer);
+	timer_delete_sync(&mc->qtimer);
+	timer_delete_sync(&mc->agingtimer);
+	timer_delete_sync(&mc->rtimer);
+	timer_delete_sync(&mc->evtimer);
 
 	return 0;
 }
@@ -2969,18 +2957,10 @@ static void mc_dev_rcu_free(struct rcu_h
 	kfree(mc);
 }
 
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 4, 0))
 static void mc_router_cleanup(struct timer_list *tl)
-#else
-static void mc_router_cleanup(unsigned long data)
-#endif
 {
 	int delay_reset = 0;
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 4, 0))
-	struct mc_struct *mc = from_timer(mc, tl, rtimer);
-#else
-	struct mc_struct *mc = (struct mc_struct *)data;
-#endif
+	struct mc_struct *mc = timer_container_of(mc, tl, rtimer);
 	unsigned long next_timer = jiffies + mc->querier_interval;
 	struct hlist_node *h;
 	struct mc_querier_entry *qe;
@@ -3024,18 +3004,10 @@ static void mc_router_cleanup(unsigned l
 	spin_unlock_bh(&mc->lock);
 }
 
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 4, 0))
 static void mc_mdb_query(struct timer_list *tl)
-#else
-static void mc_mdb_query(unsigned long data)
-#endif
 {
 	int i;
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 4, 0))
-	struct mc_struct *mc = from_timer(mc, tl, qtimer);
-#else
-	struct mc_struct *mc = (struct mc_struct *)data;
-#endif
+	struct mc_struct *mc = timer_container_of(mc, tl, qtimer);
 	unsigned long next_timer = jiffies + mc->local_query_interval;
 
 	if (mc->timeout_gmi_enable)
@@ -3079,18 +3051,10 @@ out:
 /* mc_mdb_cleanup
  *	clean up a mdb
  */
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 4, 0))
 static void mc_mdb_cleanup(struct timer_list *tl)
-#else
-static void mc_mdb_cleanup(unsigned long data)
-#endif
 {
 	int i;
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 4, 0))
-	struct mc_struct *mc = from_timer(mc, tl, agingtimer);
-#else
-	struct mc_struct *mc = (struct mc_struct *)data;
-#endif
+	struct mc_struct *mc = timer_container_of(mc, tl, agingtimer);
 	struct mc_querier_entry *igmp_root_qe = mc->rp.igmp_root_qe;
 	unsigned long next_timer, now = jiffies;
 	unsigned long igmp_expire_time = mc->membership_interval;
@@ -3257,19 +3221,11 @@ static void mc_acl_table_init(struct mc_
 /* mc_event_delay
  *	send a event out
  */
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 4, 0))
 static void mc_event_delay(struct timer_list *tl)
-#else
-static void mc_event_delay(unsigned long data)
-#endif
 {
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 4, 0))
-	struct mc_struct *mc = from_timer(mc, tl, evtimer);
+	struct mc_struct *mc = timer_container_of(mc, tl, evtimer);
 
 	mc_netlink_event_send(mc, MC_EVENT_MDB_UPDATED,	0, NULL);
-#else
-	mc_netlink_event_send((struct mc_struct *)data, MC_EVENT_MDB_UPDATED,	0, NULL);
-#endif
 }
 
 /* mc_attach
@@ -3322,17 +3278,10 @@ int mc_attach(struct net_device *dev)
 	mc->rp.type = MC_RTPORT_DEFAULT;/* If querier exist, forward IGMP/MLD message to the router port, else flood to all ports. */
 
 	mc_acl_table_init(mc);
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 4, 0))
 	timer_setup(&mc->qtimer, mc_mdb_query, 0);
 	timer_setup(&mc->agingtimer, mc_mdb_cleanup, 0);
 	timer_setup(&mc->rtimer, mc_router_cleanup, 0);
 	timer_setup(&mc->evtimer, mc_event_delay, 0);
-#else
-	setup_timer(&mc->qtimer, mc_mdb_query, (unsigned long) mc);
-	setup_timer(&mc->agingtimer, mc_mdb_cleanup, (unsigned long) mc);
-	setup_timer(&mc->rtimer, mc_router_cleanup, (unsigned long) mc);
-	setup_timer(&mc->evtimer, mc_event_delay, (unsigned long) mc);
-#endif
 	/*Make the mc ready before publishing it*/
 	if (mc_open(mc) < 0) {
 		kfree(mc);
--- a/mc_main.c
+++ b/mc_main.c
@@ -77,5 +77,6 @@ static void __exit mc_exit(void)
 
 module_init(mc_init);
 module_exit(mc_exit);
+MODULE_INFO(description, "QCA multicast snooping");
 MODULE_LICENSE("GPL v2");
 
