--- a/ecm_interface.c
+++ b/ecm_interface.c
@@ -1395,7 +1395,7 @@ __be32 ecm_interface_vxlan_gpe_get_vni_r
 		__be32 daddr;
 
 		daddr = netif_is_vxlan(indev) ? ip_hdr(skb)->saddr : ip_hdr(skb)->daddr;
-		rt = ip_route_output(&init_net, daddr, 0, 0, 0);
+		rt = ip_route_output(&init_net, daddr, 0, 0, 0, RT_SCOPE_UNIVERSE);
 		if (IS_ERR_OR_NULL(rt)) {
 			DEBUG_WARN("%px: VXLAN-GPE failed to get IPv4 route to: %pI4\n", dev, &daddr);
 			goto rt_error;
@@ -1718,7 +1718,7 @@ struct neighbour *ecm_interface_ipv4_nei
 	__be32 ipv4_addr;
 
 	ECM_IP_ADDR_TO_NIN4_ADDR(ipv4_addr, addr);
-	rt = ip_route_output(&init_net, ipv4_addr, 0, 0, 0);
+	rt = ip_route_output(&init_net, ipv4_addr, 0, 0, 0, RT_SCOPE_UNIVERSE);
 	if (IS_ERR_OR_NULL(rt)) {
 		return NULL;
 	}
@@ -9213,7 +9213,7 @@ int ecm_interface_wifi_event_stop(void)
  * ecm_interface_igs_enabled_handler()
  *	IGS enabled check sysctl node handler.
  */
-static int ecm_interface_igs_enabled_handler(struct ctl_table *ctl, int write, void __user *buffer,
+static int ecm_interface_igs_enabled_handler(const struct ctl_table *ctl, int write, void __user *buffer,
 		 size_t *lenp, loff_t *ppos)
 {
 	int ret;
@@ -9254,7 +9254,7 @@ static int ecm_interface_igs_enabled_han
  * ecm_interface_src_check_handler()
  *	Source interface check sysctl node handler.
  */
-static int ecm_interface_src_check_handler(struct ctl_table *ctl, int write, void __user *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_interface_src_check_handler(const struct ctl_table *ctl, int write, void __user *buffer, size_t *lenp, loff_t *ppos)
 {
 	int ret;
 	int current_value;
@@ -9295,7 +9295,7 @@ static int ecm_interface_src_check_handl
  * ecm_interface_src_check_no_flush_handler()
  *	Source interface check no flush sysctl node handler.
  */
-static int ecm_interface_src_check_no_flush_handler(struct ctl_table *ctl, int write, void __user *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_interface_src_check_no_flush_handler(const struct ctl_table *ctl, int write, void __user *buffer, size_t *lenp, loff_t *ppos)
 {
 	int ret;
 
@@ -9319,7 +9319,7 @@ static int ecm_interface_src_check_no_fl
  * ecm_interface_mwan3_enable_handler()
  *	 mwan3 enable check sysctl node handler.
  */
-static int ecm_interface_mwan3_enable_handler(struct ctl_table *ctl, int write, void __user *buffer,
+static int ecm_interface_mwan3_enable_handler(const struct ctl_table *ctl, int write, void __user *buffer,
 		 size_t *lenp, loff_t *ppos)
 {
 	int ret;
@@ -9657,7 +9657,7 @@ static int ecm_interface_defunct_by_mac_
  * ecm_interface_defunct_by_mac_addr_handler()
  * 	Proc handler function for defunct the ecm rules by mac address
  */
-static int ecm_interface_defunct_by_mac_address_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_interface_defunct_by_mac_address_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * To mark a MAC address as defunct:
--- a/ecm_db/ecm_db_connection.c
+++ b/ecm_db/ecm_db_connection.c
@@ -737,7 +737,7 @@ EXPORT_SYMBOL(ecm_db_connection_node_add
 void ecm_db_connection_iface_name_get(struct ecm_db_connection_instance *ci, ecm_db_obj_dir_t dir, char *name_buffer)
 {
 	DEBUG_CHECK_MAGIC(ci, ECM_DB_CONNECTION_INSTANCE_MAGIC, "%px: magic failed", ci);
-	strlcpy(name_buffer, ci->node[dir]->iface->name, IFNAMSIZ);
+	strscpy(name_buffer, ci->node[dir]->iface->name, IFNAMSIZ);
 }
 EXPORT_SYMBOL(ecm_db_connection_iface_name_get);
 
--- a/ecm_db/ecm_db_iface.c
+++ b/ecm_db/ecm_db_iface.c
@@ -250,7 +250,7 @@ static int ecm_db_iface_state_get_base(s
 	interface_identifier = ii->interface_identifier;
 	ae_interface_identifier = ii->ae_interface_identifier;
 	spin_lock_bh(&ecm_db_lock);
-	strlcpy(name, ii->name, IFNAMSIZ);
+	strscpy(name, ii->name, IFNAMSIZ);
 	mtu = ii->mtu;
 	spin_unlock_bh(&ecm_db_lock);
 
@@ -1241,7 +1241,7 @@ void ecm_db_iface_interface_name_get(str
 {
 	DEBUG_CHECK_MAGIC(ii,
 		ECM_DB_IFACE_INSTANCE_MAGIC, "%px: magic failed", ii);
-	strlcpy(name_buffer, ii->name, IFNAMSIZ);
+	strscpy(name_buffer, ii->name, IFNAMSIZ);
 }
 EXPORT_SYMBOL(ecm_db_iface_interface_name_get);
 
@@ -2905,7 +2905,7 @@ void ecm_db_iface_add_ethernet(struct ec
 #endif
 	ii->arg = arg;
 	ii->final = final;
-	strlcpy(ii->name, name, IFNAMSIZ);
+	strscpy(ii->name, name, IFNAMSIZ);
 	ii->mtu = mtu;
 	ii->interface_identifier = interface_identifier;
 	ii->ae_interface_identifier = ae_interface_identifier;
@@ -2957,7 +2957,7 @@ void ecm_db_iface_add_dsa(struct ecm_db_
 #endif
 	ii->arg = arg;
 	ii->final = final;
-	strlcpy(ii->name, name, IFNAMSIZ);
+	strscpy(ii->name, name, IFNAMSIZ);
 	ii->mtu = mtu;
 	ii->interface_identifier = interface_identifier;
 	ii->ae_interface_identifier = ae_interface_identifier;
@@ -3008,7 +3008,7 @@ void ecm_db_iface_add_lag(struct ecm_db_
 #endif
 	ii->arg = arg;
 	ii->final = final;
-	strlcpy(ii->name, name, IFNAMSIZ);
+	strscpy(ii->name, name, IFNAMSIZ);
 	ii->mtu = mtu;
 	ii->interface_identifier = interface_identifier;
 	ii->ae_interface_identifier = ae_interface_identifier;
@@ -3059,7 +3059,7 @@ void ecm_db_iface_add_bridge(struct ecm_
 #endif
 	ii->arg = arg;
 	ii->final = final;
-	strlcpy(ii->name, name, IFNAMSIZ);
+	strscpy(ii->name, name, IFNAMSIZ);
 	ii->mtu = mtu;
 	ii->interface_identifier = interface_identifier;
 	ii->ae_interface_identifier = ae_interface_identifier;
@@ -3110,7 +3110,7 @@ void ecm_db_iface_add_ovs_bridge(struct
 #endif
 	ii->arg = arg;
 	ii->final = final;
-	strlcpy(ii->name, name, IFNAMSIZ);
+	strscpy(ii->name, name, IFNAMSIZ);
 	ii->mtu = mtu;
 	ii->interface_identifier = interface_identifier;
 	ii->ae_interface_identifier = ae_interface_identifier;
@@ -3159,7 +3159,7 @@ void ecm_db_iface_add_ovs_internal(struc
 #endif
 	ii->arg = arg;
 	ii->final = final;
-	strlcpy(ii->name, name, IFNAMSIZ);
+	strscpy(ii->name, name, IFNAMSIZ);
 	ii->mtu = mtu;
 	ii->interface_identifier = interface_identifier;
 	ii->ae_interface_identifier = ae_interface_identifier;
@@ -3210,7 +3210,7 @@ void ecm_db_iface_add_macvlan(struct ecm
 #endif
 	ii->arg = arg;
 	ii->final = final;
-	strlcpy(ii->name, name, IFNAMSIZ);
+	strscpy(ii->name, name, IFNAMSIZ);
 	ii->mtu = mtu;
 	ii->interface_identifier = interface_identifier;
 	ii->ae_interface_identifier = ae_interface_identifier;
@@ -3261,7 +3261,7 @@ void ecm_db_iface_add_vlan(struct ecm_db
 #endif
 	ii->arg = arg;
 	ii->final = final;
-	strlcpy(ii->name, name, IFNAMSIZ);
+	strscpy(ii->name, name, IFNAMSIZ);
 	ii->mtu = mtu;
 	ii->interface_identifier = interface_identifier;
 	ii->ae_interface_identifier = ae_interface_identifier;
@@ -3315,7 +3315,7 @@ void ecm_db_iface_add_map_t(struct ecm_d
 #endif
 	ii->arg = arg;
 	ii->final = final;
-	strlcpy(ii->name, name, IFNAMSIZ);
+	strscpy(ii->name, name, IFNAMSIZ);
 	ii->mtu = mtu;
 	ii->interface_identifier = interface_identifier;
 	ii->ae_interface_identifier = ae_interface_identifier;
@@ -3367,7 +3367,7 @@ void ecm_db_iface_add_gre_tun(struct ecm
 #endif
 	ii->arg = arg;
 	ii->final = final;
-	strlcpy(ii->name, name, IFNAMSIZ);
+	strscpy(ii->name, name, IFNAMSIZ);
 	ii->mtu = mtu;
 	ii->interface_identifier = interface_identifier;
 	ii->ae_interface_identifier = ae_interface_identifier;
@@ -3419,7 +3419,7 @@ void ecm_db_iface_add_pppoe(struct ecm_d
 #endif
 	ii->arg = arg;
 	ii->final = final;
-	strlcpy(ii->name, name, IFNAMSIZ);
+	strscpy(ii->name, name, IFNAMSIZ);
 	ii->mtu = mtu;
 	ii->interface_identifier = interface_identifier;
 	ii->ae_interface_identifier = ae_interface_identifier;
@@ -3472,7 +3472,7 @@ void ecm_db_iface_add_pppol2tpv2(struct
 #endif
 	ii->arg = arg;
 	ii->final = final;
-	strlcpy(ii->name, name, IFNAMSIZ);
+	strscpy(ii->name, name, IFNAMSIZ);
 	ii->mtu = mtu;
 	ii->interface_identifier = interface_identifier;
 	ii->ae_interface_identifier = ae_interface_identifier;
@@ -3525,7 +3525,7 @@ void ecm_db_iface_add_pptp(struct ecm_db
 #endif
 	ii->arg = arg;
 	ii->final = final;
-	strlcpy(ii->name, name, IFNAMSIZ);
+	strscpy(ii->name, name, IFNAMSIZ);
 	ii->mtu = mtu;
 	ii->interface_identifier = interface_identifier;
 	ii->ae_interface_identifier = ae_interface_identifier;
@@ -3575,7 +3575,7 @@ void ecm_db_iface_add_unknown(struct ecm
 #endif
 	ii->arg = arg;
 	ii->final = final;
-	strlcpy(ii->name, name, IFNAMSIZ);
+	strscpy(ii->name, name, IFNAMSIZ);
 	ii->mtu = mtu;
 	ii->interface_identifier = interface_identifier;
 	ii->ae_interface_identifier = ae_interface_identifier;
@@ -3624,7 +3624,7 @@ void ecm_db_iface_add_loopback(struct ec
 #endif
 	ii->arg = arg;
 	ii->final = final;
-	strlcpy(ii->name, name, IFNAMSIZ);
+	strscpy(ii->name, name, IFNAMSIZ);
 	ii->mtu = mtu;
 	ii->interface_identifier = interface_identifier;
 	ii->ae_interface_identifier = ae_interface_identifier;
@@ -3683,7 +3683,7 @@ void ecm_db_iface_add_sit(struct ecm_db_
 #endif
 	ii->arg = arg;
 	ii->final = final;
-	strlcpy(ii->name, name, IFNAMSIZ);
+	strscpy(ii->name, name, IFNAMSIZ);
 	ii->mtu = mtu;
 	ii->interface_identifier = interface_identifier;
 	ii->ae_interface_identifier = ae_interface_identifier;
@@ -3733,7 +3733,7 @@ void ecm_db_iface_add_tunipip6(struct ec
 #endif
 	ii->arg = arg;
 	ii->final = final;
-	strlcpy(ii->name, name, IFNAMSIZ);
+	strscpy(ii->name, name, IFNAMSIZ);
 	ii->mtu = mtu;
 	ii->interface_identifier = interface_identifier;
 	ii->ae_interface_identifier = ae_interface_identifier;
@@ -3889,7 +3889,7 @@ void ecm_db_iface_add_ovpn(struct ecm_db
 #endif
 	ii->arg = arg;
 	ii->final = final;
-	strlcpy(ii->name, name, IFNAMSIZ);
+	strscpy(ii->name, name, IFNAMSIZ);
 	ii->mtu = mtu;
 	ii->interface_identifier = interface_identifier;
 	ii->ae_interface_identifier = type_info->tun_ifnum;
@@ -3940,7 +3940,7 @@ void ecm_db_iface_add_vxlan(struct ecm_d
 #endif
 	ii->arg = arg;
 	ii->final = final;
-	strlcpy(ii->name, name, IFNAMSIZ);
+	strscpy(ii->name, name, IFNAMSIZ);
 	ii->mtu = mtu;
 	ii->interface_identifier = interface_identifier;
 	ii->ae_interface_identifier = ae_interface_identifier;
--- a/ecm_tracker_udp.c
+++ b/ecm_tracker_udp.c
@@ -1160,7 +1160,7 @@ static int ecm_tracker_rtp_payload_types
  * ecm_tracker_udp_rtp_payload_types_handler()
  *	Proc handler function for RTP payload types read/write operation.
  */
-static int ecm_tracker_udp_rtp_payload_types_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_tracker_udp_rtp_payload_types_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
@@ -1180,7 +1180,7 @@ static int ecm_tracker_udp_rtp_payload_t
  * ecm_tracker_udp_clf_enabled_handler()
  *	Sysctl to enable/disable UDP classifier flag.
  */
-int ecm_tracker_udp_clf_enabled_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+int ecm_tracker_udp_clf_enabled_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	int ret;
 
--- a/frontends/ecm_front_end_common.c
+++ b/frontends/ecm_front_end_common.c
@@ -18,6 +18,7 @@
  */
 
 #include <linux/version.h>
+#include <linux/vmalloc.h>
 #include <linux/module.h>
 #include <linux/of.h>
 #include <linux/debugfs.h>
@@ -947,7 +948,7 @@ int ecm_front_end_ppe_fse_enable_handler
  * ecm_front_end_db_conn_limit_handler()
  *	Database connection limit sysctl node handler.
  */
-int ecm_front_end_db_conn_limit_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+int ecm_front_end_db_conn_limit_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	int ret;
 	int current_value;
@@ -1166,7 +1167,7 @@ static int ecm_front_end_denied_ports_ha
  * ecm_front_end_udp_denied_ports_handler()
  *	Proc handler function for UDP denied ports read/write operation.
  */
-static int ecm_front_end_udp_denied_ports_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+int ecm_front_end_udp_denied_ports_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
@@ -1186,7 +1187,7 @@ static int ecm_front_end_udp_denied_port
  * ecm_front_end_tcp_denied_ports_handler()
  *	Proc handler function for TCP denied ports read/write operation.
  */
-static int ecm_front_end_tcp_denied_ports_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+int ecm_front_end_tcp_denied_ports_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
--- a/frontends/nss/ecm_nss_multicast_ipv6.c
+++ b/frontends/nss/ecm_nss_multicast_ipv6.c
@@ -1010,6 +1010,7 @@ static void ecm_nss_multicast_ipv6_conne
 			create->ingress_vlan_tag[interface_type_counts[ii_type]] = ((vlan_info.vlan_tpid << 16) | vlan_info.vlan_tag);
 			create->valid_flags |= NSS_IPV6_MC_RULE_CREATE_FLAG_INGRESS_VLAN_VALID;
 			DEBUG_TRACE("%px: vlan tag: %x\n", feci, create->ingress_vlan_tag[interface_type_counts[ii_type]]);
+			fallthrough;
 #else
 			rule_invalid = true;
 			DEBUG_TRACE("%px: VLAN - unsupported\n", feci);
--- a/frontends/sfe/ecm_sfe_common.c
+++ b/frontends/sfe/ecm_sfe_common.c
@@ -347,7 +347,7 @@ void ecm_sfe_common_fast_xmit_set(uint32
  * ecm_sfe_fast_xmit_enable_handler()
  *	Fast transmit sysctl node handler.
  */
-int ecm_sfe_fast_xmit_enable_handler(struct ctl_table *ctl, int write, void __user *buffer, size_t *lenp, loff_t *ppos)
+int ecm_sfe_fast_xmit_enable_handler(const struct ctl_table *ctl, int write, void __user *buffer, size_t *lenp, loff_t *ppos)
 {
 	int ret;
 
@@ -371,7 +371,7 @@ int ecm_sfe_fast_xmit_enable_handler(str
  * ecm_sfe_fse_enable_handler()
  *	Sysctl to enable/disable FSE programming through ECM SFE frontend.
  */
-int ecm_sfe_fse_enable_handler(struct ctl_table *ctl, int write, void __user *buffer, size_t *lenp, loff_t *ppos)
+int ecm_sfe_fse_enable_handler(const struct ctl_table *ctl, int write, void __user *buffer, size_t *lenp, loff_t *ppos)
 {
 	int ret;
 	int current_val;
@@ -543,7 +543,7 @@ static struct ctl_table ecm_sfe_sysctl_t
  * ecm_sfe_sysctl_tbl_init()
  * 	Register sysctl for SFE
  */
-int ecm_sfe_sysctl_tbl_init()
+int ecm_sfe_sysctl_tbl_init(void)
 {
 	ecm_sfe_ctl_tbl_hdr = register_sysctl(ECM_FRONT_END_SYSCTL_PATH, ecm_sfe_sysctl_tbl);
 	if (!ecm_sfe_ctl_tbl_hdr) {
--- a/ecm_classifier_hyfi.c
+++ b/ecm_classifier_hyfi.c
@@ -437,7 +437,7 @@ static void ecm_classifier_hyfi_get_brid
 			struct net_device *master;
 			master = ecm_interface_get_and_hold_dev_master(dev);
 			if (master) {
-				strlcpy(name_buffer, master->name, IFNAMSIZ);
+				strscpy(name_buffer, master->name, IFNAMSIZ);
 				dev_put(master);
 				dev_put(dev);
 				break;
@@ -841,13 +841,13 @@ struct ecm_classifier_hyfi_instance *ecm
 		/* one of the bridge name is null, typical
 		 * routed connection. Consider valid bridge*/
 		if (strlen(to_bridge)) {
-			strlcpy(chfi->bridge_name, to_bridge, IFNAMSIZ);
+			strscpy(chfi->bridge_name, to_bridge, IFNAMSIZ);
 		} else if (strlen(from_bridge)) {
-			strlcpy(chfi->bridge_name, from_bridge, IFNAMSIZ);
+			strscpy(chfi->bridge_name, from_bridge, IFNAMSIZ);
 		}
 	} else if (!strncmp(to_bridge, from_bridge, IFNAMSIZ)) {
 		/* Pure bridge connection. Consider any one bridge */
-		strlcpy(chfi->bridge_name, to_bridge, IFNAMSIZ);
+		strscpy(chfi->bridge_name, to_bridge, IFNAMSIZ);
 	} else {
 		/* multi-bridge connection */
 		chfi->multi_bridge_flow = true;
--- a/examples/ecm_pcc_test.c
+++ b/examples/ecm_pcc_test.c
@@ -716,9 +716,9 @@ static unsigned int ecm_pcc_test_update_
 	o_feature_flags = rule->feature_flags;
 	rule->accel = accel;
 	rule->feature_flags = feature_flags;
-	strlcpy(rule->name, name, sizeof(rule->name));
-	strlcpy(rule->mirror_info.tuple_mirror_dev, tuple_mirror_dev, IFNAMSIZ);
-	strlcpy(rule->mirror_info.tuple_ret_mirror_dev, tuple_ret_mirror_dev, IFNAMSIZ);
+	strscpy(rule->name, name, sizeof(rule->name));
+	strscpy(rule->mirror_info.tuple_mirror_dev, tuple_mirror_dev, IFNAMSIZ);
+	strscpy(rule->mirror_info.tuple_ret_mirror_dev, tuple_ret_mirror_dev, IFNAMSIZ);
 	rule->ap_info.flow_ap_index = flow_ap_index;
 	rule->ap_info.return_ap_index = return_ap_index;
 	spin_unlock_bh(&ecm_pcc_test_rules_lock);
@@ -881,7 +881,7 @@ static unsigned int ecm_pcc_test_add_rul
 	if (!new_rule)
 		return 0;
 
-	strlcpy(new_rule->name, name, sizeof(new_rule->name));
+	strscpy(new_rule->name, name, sizeof(new_rule->name));
 	new_rule->accel = accel;
 	new_rule->proto = proto;
 	new_rule->src_port = src_port;
@@ -892,8 +892,8 @@ static unsigned int ecm_pcc_test_add_rul
 	new_rule->dest_addr = *dest_addr;
 	new_rule->ipv = ipv;
 	new_rule->feature_flags = feature_flags;
-	strlcpy(new_rule->mirror_info.tuple_mirror_dev, tuple_mirror_dev, IFNAMSIZ);
-	strlcpy(new_rule->mirror_info.tuple_ret_mirror_dev, tuple_ret_mirror_dev, IFNAMSIZ);
+	strscpy(new_rule->mirror_info.tuple_mirror_dev, tuple_mirror_dev, IFNAMSIZ);
+	strscpy(new_rule->mirror_info.tuple_ret_mirror_dev, tuple_ret_mirror_dev, IFNAMSIZ);
 	new_rule->ap_info.flow_ap_index = flow_ap_index;
 	new_rule->ap_info.return_ap_index = return_ap_index;
 	INIT_LIST_HEAD(&new_rule->list);
@@ -1024,7 +1024,7 @@ static ssize_t ecm_pcc_test_rule_write(s
 	/*
 	 * Convert fields
 	 */
-	strlcpy(name, fields[0], sizeof(name));
+	strscpy(name, fields[0], sizeof(name));
 
 	name[sizeof(name) - 1] = 0;
 	if (sscanf(fields[1], "%u", &oper) != 1)
@@ -1072,9 +1072,9 @@ static ssize_t ecm_pcc_test_rule_write(s
 		return -EINVAL;
 	}
 
-	strlcpy(tuple_mirror_dev, fields[11], IFNAMSIZ);
+	strscpy(tuple_mirror_dev, fields[11], IFNAMSIZ);
 
-	strlcpy(tuple_ret_mirror_dev, fields[12], IFNAMSIZ);
+	strscpy(tuple_ret_mirror_dev, fields[12], IFNAMSIZ);
 
 	if (sscanf(fields[13], "%d", &flow_ap_index) != 1)
 		goto sscanf_read_error;
--- a/Makefile
+++ b/Makefile
@@ -567,4 +567,7 @@ endif
 
 ccflags-y += -Wall -Werror
 
+# GCC 15+ has stricter checks for prototypes and declarations. Silence them
+ccflags-y += -Wno-missing-prototypes -Wno-missing-declarations
+
 obj ?= .
